sklearn/_config.py

```diff
@@ -12,6 +12,7 @@
     "pairwise_dist_chunk_size": int(
         os.environ.get("SKLEARN_PAIRWISE_DIST_CHUNK_SIZE", 256)
     ),
+    "enable_cython_pairwise_dist": True,
```

Comment:
Are we going to keep this or is it a temporary config for a few releases until we can finally get rid of the old code path?
I'd like to avoid an unnecessary deprecation cycle.

Reply:
According to its documentation, it's just for benchmarking and debugging purpose. It's ok to keep it while the PR is in development, but I'd be in favor of removing it before merging to into main.

Reply:
This is only needed for a test. I would likely remove it.

Reply:
Fine with me. Then let us try to not make it public and to not forget it before merging in main.

Reply:
I would rather keep it, at least for the next release or too. Once the old numpy route is completely removed we can deprecate the option. But that will require implementing Cython support for the sparse and float32 data-structure which I would rather not wait for to merge to `main` or release 1.1.

Reply:
> I would likely not document it to avoid a deprecation cycle.

I prefer to document something as maintainer-only than not documenting it at all.

Reply:
> This is only needed for test_kneighbors_brute_backend

But the purpose of this test is to test that the new code gives the same result as the old one. Although I think it's really important to have that during the development, I don't think it's a test we want in the code base, since the end goal is to not have the old backend.

So would it make sense to remove these debugging parts before merging the final PR in main ?

Reply:
The old back-end won't be removed as part of https://github.com/scikit-learn/scikit-learn/pull/22134.
Hence, I think is better to keep this test as long as we still have the old back-end.